src/test/unit/resources/test_detectors.py

```diff
@@ -54,10 +55,49 @@ def test_env_vars_detector():
 def test_get_resource_aws_ecs_resource_detector():
     os.environ["ECS_CONTAINER_METADATA_URI"] = "mock-url"
 
-    resource = get_resource({"a": "b"})
+    resource = get_resource({})
 
     assert resource.attributes["cloud.provider"] == "aws"
     assert resource.attributes["cloud.platform"] == "aws_ecs"
     assert isinstance(resource.attributes["container.name"], str)
     assert len(resource.attributes["container.name"]) > 1
     assert isinstance(resource.attributes["container.id"], str)
+
+
+def load_json(filename: str):
+    with open(os.path.join(os.path.dirname(__file__), filename)) as f:
+        return json.load(f)
+
+
+@responses.activate
```

Comment:
1. can you add a test to see how we handle exceptions
2. can you add timeout mechanism for the requests sent to `METADATA_URI` ?

Reply:
2. It is done by passing `timeout` to `urlopen`.